### PR TITLE
Update "Distributing a library"

### PR DESCRIPTION
**Pull request by kwalrath**
Followup to #1054 (and then some).

It started out with me fixing a typo. Then it grew...

Staged at: https://kw-www-dartlang-1.firebaseapp.com/guides/libraries/create-library-packages#distributing-an-open-source-library

/cc @kevmoo @Sfshaza

### CODE REVIEW

File: src/_guides/libraries/create-library-packages.md [hunk 1]
```diff
@@ -211,38 +211,38 @@ For an example of generated docs, see the
...
+For example, the page for the shelf package is at
+[pub.dartlang.org/packages/shelf](https://pub.dartlang.org/packages/shelf).
+See [Publishing a Package](/tools/pub/publishing)
+for details on how to prepare your package for publishing.
+
+The pub site not only hosts your package,
+but also generates and hosts your package's API reference docs.
+A link to the latest generated docs is in the package's **About** box;
+for example, the shelf package's **About** box links to the API docs at the URL
 [https://pub.dartlang.org/documentation/shelf/latest/](https://pub.dartlang.org/documentation/shelf/latest/).
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_guides/libraries/create-library-packages.md]: I meant to emphasize the "latest" (which was the subject of a separate note that I deleted), but it looks like that isn't necessary, now that the version string isn't used for the "latest" version's URLs. (We should do the same on api.dartlang.org!)

File: src/_guides/libraries/create-library-packages.md [hunk 2]
```diff
@@ -211,38 +211,38 @@ For an example of generated docs, see the
...
+## Distributing an open source library {#distributing-a-library}
+
+If your library is open source,
+we recommend sharing it on the pub site,
+[pub.dartlang.org.](https://pub.dartlang.org/)
+To publish or update the library,
+use [pub publish](/tools/pub/cmd/pub-lish),
+which uploads your package and creates or updates its page.
+For example, the page for the shelf package is at
+[pub.dartlang.org/packages/shelf](https://pub.dartlang.org/packages/shelf).
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_guides/libraries/create-library-packages.md]: ?

File: src/_guides/libraries/create-library-packages.md [hunk 3]
```diff
@@ -263,3 +263,5 @@ Use the following resources to learn more about library packages:
   [shelf](https://github.com/dart-lang/shelf),
   [source_gen](https://github.com/dart-lang/source_gen), and
   [test](https://github.com/dart-lang/test).
+
+[dartdoc]: https://github.com/dart-lang/dartdoc#dartdoc)
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_guides/libraries/create-library-packages.md]: Oops!

File: src/_guides/libraries/create-library-packages.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_guides/libraries/create-library-packages.md]: Ah, never mind, I can see what you meant by editing your comment.

File: src/_guides/libraries/create-library-packages.md [hunk 4]
```diff
@@ -211,38 +211,38 @@ For an example of generated docs, see the
...
-
-To minimize the possibility of broken links as version numbers change,
-specify "latest" rather than a specific version when linking to
-documentation on [pub.dartlang.org](https://pub.dartlang.org/). 
-For example:
+## Distributing an open source library {#distributing-a-library}
+
+If your library is open source,
+we recommend sharing it on the pub site,
+[pub.dartlang.org.](https://pub.dartlang.org/)
```
[the remark this answers is not in the document]
Comment (chalin) [reply src/_guides/libraries/create-library-packages.md]: Oh, did you see the `{{site.pub}}` in the original comment? (Sry, I didn't check how it was rendered.)

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_guides/libraries/create-library-packages.md]: Yes, I saw that once I realized that the following comment had to be missing something.

File: src/_guides/libraries/create-library-packages.md [hunk 5]
```diff
@@ -211,38 +211,38 @@ For an example of generated docs, see the
...
+which uploads your package and creates or updates its page.
+For example, the page for the shelf package is at
+[pub.dartlang.org/packages/shelf](https://pub.dartlang.org/packages/shelf).
+See [Publishing a Package](/tools/pub/publishing)
+for details on how to prepare your package for publishing.
+
+The pub site not only hosts your package,
+but also generates and hosts your package's API reference docs.
+A link to the latest generated docs is in the package's **About** box;
+for example, the shelf package's **About** box links to the API docs at the URL
```
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_guides/libraries/create-library-packages.md]: Ah, I remember why I didn't do this. I was hoping to change some of the strings in the UI, and I think this was one of them.

**Pull request merged**